src/System.Net.Http/src/System/Net/Http/StreamToStreamCopy.cs

```diff
@@ -2,59 +2,266 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Diagnostics.Contracts;
+using System.Diagnostics;
 using System.IO;
 using System.Threading.Tasks;
 
 namespace System.Net.Http
 {
-    // This helper class is used to copy the content of a source stream to a destination stream.
-    // The type verifies if the source and/or destination stream are MemoryStreams (or derived types). If so, sync
-    // read/write is used on the MemoryStream to avoid context switches.
+    /// <summary>
+    /// Helper class is used to copy the content of a source stream to a destination stream,
+    /// with optimizations based on expected usage within HttpClient and with the ability
+    /// to dispose of the source stream when the copy has completed.
+    /// </summary>
     internal static class StreamToStreamCopy
     {
-        public static async Task CopyAsync(Stream source, Stream destination, int bufferSize, bool disposeSource)
+        /// <summary>Copies the source stream from its current position to the destination stream at its current position.</summary>
+        /// <param name="source">The source stream from which to copy.</param>
+        /// <param name="destination">The destination stream to which to copy.</param>
+        /// <param name="bufferSize">The size of the buffer to allocate if one needs to be allocated.</param>
+        /// <param name="disposeSource">Whether to dispose of the source stream after the copy has finished successfully.</param>
```

Comment:
Are XML comments the 'new' thing to do in CoreFx sources?  Have we updated the coding guidelines documents for this and do we have now have tooling that uses this?